pystac/extensions/eo.py

```diff
@@ -468,6 +471,10 @@ class SummariesEOExtension(SummariesExtension):
     defined in the :stac-ext:`Electro-Optical Extension <eo>`.
     """
 
+    def __init__(self, collection: pystac.Collection):
+        EOExtension.add_to(collection)
+        super().__init__(collection)
+
```

Comment:
I think it might be better to put this logic into `EOExtension.summaries` and have that method behave similarly to [`EOExtension.ext`](https://github.com/stac-utils/pystac/blob/12eff70f9423d642c9909c92b4ba228bf97bef0e/pystac/extensions/eo.py#L350):
1. Add an `add_if_missing: bool = False` argument to `EOExtension.summaries`
2. If `add_if_missing` is `False`, raise an exception if someone tries to get summaries for a Collection that is missing the schema URI.
3. Add the schema URI, if necessary, when `add_is_missing` is `True`

The reasoning would be similar to that for the `ext` method (see [this discussion](https://github.com/stac-utils/stactools/pull/113#discussion_r639860502) for context). Basically, in cases where someone is using PySTAC for reading existing Collections it should let them know the extension is not implemented rather than fixing it behind the scenes. If someone really does want to fix it automatically, then `add_if_missing` gives them an easy but explicit way to do that.

@lossyrob @gadomski What are your thoughts on this?

This would obviously need to be applied to other extension with this pattern, too, if we do decide to go this route.

Reply:
I've been liking the `add_if_missing` pattern; I can omit it when testing to get trigger an error on a missing schema, but it's still a one-liner to get an extension in (e.g.) `Item` creation. So I'm :+1: on @duckontheweb's proposal.

Reply:
+1

Reply:
Thanks for the links, I'm making this change this morning

Reply:
I think I've faithfully followed this suggestion in 6b4413f -- lemme know if that's not what was intended, if it's consistent with what's requested I'll copy-🍝 my way through the other extensions

Reply:
![](https://media.giphy.com/media/zCME2Cd20Czvy/giphy.gif)